tests/contracts/matchers.nim

```diff
@@ -0,0 +1,26 @@
+import std/json
+import std/strutils
+import pkg/asynctest
+import pkg/ethers
+
+proc revertReason*(e: ref ValueError): string =
+  try:
+    let json = parseJson(e.msg)
+    var msg = json{"message"}.getStr
+    const revertPrefix =
+      "Error: VM Exception while processing transaction: reverted with " &
+      "reason string "
+    msg = msg.replace(revertPrefix)
+    msg = msg.replace("\'", "")
+    return msg
+  except JsonParsingError:
+    return ""
+
+
+template revertsWith*(reason, body) =
+  var revertReason = ""
+  try:
+    body
+  except ValueError as e:
+    revertReason = e.revertReason
+  check revertReason == reason
```

Comment:
I was considering putting this in `nim-ethers`, thoughts?

Reply:
I was not 100% sure where the `ValueError` comes from. I don't believe `nim-ethers` raised the error otherwise it likely would have been a `JsonRpcProviderError`.

Reply:
Created a PR for the changes in ethers (https://github.com/status-im/nim-ethers/pull/33), but have not updated this PR to use those changes yet. Will do tomorrow.